cloudwatch/helpers.go

```diff
@@ -72,8 +73,31 @@ func parseKeysTemplate(data map[interface{}]interface{}, keys string, w io.Write
 // from an interface{} map (expected to contain strings or more interface{} maps).
 // This runs once for every log line.
 // Used to fill in any template variables that may exist in the logStream or logGroup names.
-func parseDataMapTags(e *Event, logTags []string, t *fastTemplate, w io.Writer) (int64, error) {
+func parseDataMapTags(e *Event, logTags []string, t *fastTemplate, metadata TaskMetadata, uuid string, w io.Writer) (int64, error) {
```

Comment:
We're probably at a point where it makes sense to convert this procedure to a method on the `OutputPlugin` struct. Then you don't ned to pass these three `output.` variables in here at all.